python/pyarrow/io.pxi

```diff
@@ -262,6 +271,158 @@ cdef class NativeFile:
         Alias for read, needed to match the IOBase interface."""
         return self.read(nbytes=None)
 
+    def readall(self):
+        return self.read()
+
+    def readinto(self, b):
+        """
+        Read into the supplied buffer
+
+        Parameters
+        -----------
+        b: any python object supporting buffer interface
+
+        Returns
+        --------
+        number of bytes written
+        """
+
+        cdef:
+            int64_t bytes_read
+            uint8_t* buf
+
+        self._assert_readable()
+
+        cdef Buffer py_buf = py_buffer(b)
+        cdef int64_t buf_len = py_buf.size
+
+        buf = py_buf.buffer.get().mutable_data()
+
+        with nogil:
+            check_status(self.rd_file.get().Read(buf_len, &bytes_read, buf))
+
+        return bytes_read
+
+    def readline(self, size=None):
```

Comment:
Ok... I really don't like that we're reinventing our own suboptimal readline implementation, just because Pandas somewhat requires it. @wesm do you think there's a way to please Pandas without doing this? Perhaps raise NotImplementedError?